velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,127 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
+class StringToMap final : public exec::VectorFunction {
+ public:
+  StringToMap(char entryDelim, char keyValueDelim)
+      : entryDelim_(entryDelim), keyValueDelim_(keyValueDelim) {}
+
+  void apply(
```

Comment:
What's the motivation for writing this as a VectorFunction and not using Simple Function API?

Reply:
@mbasmanova, thanks for your review!
Could you give me some advice about when to use vector function and when to use simple function? I am assuming vector function generally performs better, but using simple function API seems more common in velox.

Reply:
Simple Function API should be preferred unless VectorFunction implementation provides a significant performance gain which needs to be demonstrated with a benchmark.

Reply:
Hi @mbasmanova, sorry for this late update. I have designed a simple benchmark. According to the below result, the VectorFunction (1st line) proposed in this PR performs worse than a simple function (2nd line). Should I re-implement it with simple function API?

```txt
============================================================================
[...]hmarks/ExpressionBenchmarkBuilder.cpp     relative  time/iter   iters/s
============================================================================
##str_to_map (vector API)                                    1.89s   529.39m
##split_to_map (simple API)                                  1.56s   640.34m

```
Here, the second line is the result of velox's `split_to_map` presto function, which has different handling from spark `str_to_map` on some corner cases . But in this benchmark, only some regular cases are tested.

Reply:
@PHILO-HE Yes, please, change to simple function.